### PR TITLE
feat: Lightning channels API

### PR DESCRIPTION
**Pull request by michael1011**
Precursor for #197

### CODE REVIEW

File: boltzr/src/api/lightning.rs [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    S: SwapInfos + Send + Sync + Clone + 'static,
+    M: SwapManager + Send + Sync + 'static,
+{
+    let cln = state
+        .manager
+        .get_currency(&currency)
+        .and_then(|currency| currency.cln);
+
+    Ok(match cln {
+        Some(mut cln) => {
```
Comment (maybeast) [inline boltzr/src/api/lightning.rs:48]: what happens in case of 0 amount?

File: boltzr/src/api/lightning.rs [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+                    network: Network::Regtest,
+                    wallet: Arc::new(crate::wallet::Bitcoin::new(Network::Regtest)),
+                    cln: Some(cln.clone()),
+                    lnd: None,
+                    chain: None,
+                })
+            });
+        }
+
+        let amount = 21;
```
Comment (maybeast) [inline boltzr/src/api/lightning.rs:205]: want to add a test for 0 amount and overflow?

File: boltzr/src/service/lightning_info.rs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+                return Ok(channels);
+            }
+        }
+
+        let mut cln = self.get_cln(symbol)?;
+        let raw_channels: Vec<ListchannelsChannels> = cln.list_channels(Some(destination)).await?;
+
+        let mut channels = Vec::new();
+        for channel in raw_channels {
+            if !channel.public {
```
Comment (maybeast) [inline boltzr/src/service/lightning_info.rs:177]: want to add a test for this?

File: boltzr/src/service/lightning_info.rs [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    currencies: Currencies,
+    cache: Option<C>,
+}
+
+impl<C: Cache + Send + Sync> ClnLightningInfo<C> {
+    pub fn new(cache: Option<C>, currencies: Currencies) -> Self {
+        Self { cache, currencies }
+    }
+
+    async fn get_node_info_from_cln(&self, cln: &mut Cln, id: Vec<u8>) -> anyhow::Result<Node> {
```
Comment (jackstar12) [inline boltzr/src/service/lightning_info.rs:107]: Why have this extra function? `InfoFetchError` implements `From` for anyhow, so we could put it all in the `get_node_info` implementation. A bit confusing this way

File: boltzr/src/service/lightning_info.rs (hunk 3 above)
Comment (michael1011) [reply boltzr/src/service/lightning_info.rs:177]: Funnily enough, we can't. I can' create a scenario where `public` is false. I just added that if just in case

File: boltzr/src/api/lightning.rs (hunk 1 above)
Comment (michael1011) [reply boltzr/src/api/lightning.rs:48]: An invoice wiht 0 amount is fetched. We should not allowed that

File: boltzr/src/api/lightning.rs (hunk 2 above)
Comment (michael1011) [reply boltzr/src/api/lightning.rs:205]: I am gonna forbid 0 amount and add a test for that. Testing overflow is testing serde which I really don't want to do

**Review by maybeast — APPROVED** [review]
utACK
